spec/fixtures/discriminator_json_schema.json

```diff
@@ -1,51 +1,51 @@
 {
   "schema": {
     "discriminator": {
-      "propertyName": "key",
-      "mapping": {
-        "value1": {
-          "schema": {
-            "type": "object",
-            "required": ["data"],
-            "properties": {
-              "data": {
-                "type": "object",
-                "required": ["example_one"],
-                "properties": {
-                  "example_one": {
-                    "type": "integer",
-                    "exclusiveMinimum": 0
-                  }
+      "propertyName": "key"
+    },
+    "properties": {
+      "key": {
+        "type": "string",
+        "enum": ["value1", "value2"]
+      }
+    },
+    "anyOf": [
+      {
+        "schema": {
```

Comment:
I think we can also update `spec/fixtures/mapping-value-one.yml` to use `value1` as the top level property instead of `schema` such that it's clearer that it's dependent on the discriminator property value. And update the README to explain the usage a bit more. WDYT? 🤔

Reply:
Hmm, you're right. In the client side, we need a way to identify the this schema matches the `propertyName` value. Normally, they use the schema ref to identify that. Here, we don't have such value, so we need to see how the industry identifies these values.

Reply:
Ok, I looked into OpenAPI spec and see that the proper solution for this is [bundling instead of dereferecing](https://github.com/Redocly/redoc/issues/862#issuecomment-491631228). Here is an example: given that we have  a schema like this

```yaml
components:
  main:
    schema:
      discriminator:
        propertyName: key
      anyOf:
      - $ref: 'mapping-value-one.yml'
      - $ref: 'mapping-value-two.yml'
      properties:
        key:
          type: string
          enum:
            - value1
            - value2
```

- dereferencing will result in:

```json
{
  "components": {
    "main": {
      "schema": {
        "discriminator": {
          "propertyName": "key"
        },
        "properties": {
          "key": {
            "type": "string",
            "enum": ["value1", "value2"]
          }
        },
        "anyOf": [
          {
            "schema": {
              "type": "object",
              "required": ["data"],
              "properties": {
                "data": {
                  "type": "object",
                  "required": ["example_one"],
                  "properties": {
                    "example_one": {
                      "type": "integer",
                      "exclusiveMinimum": 0
                    }
                  }
                }
              }
            }
          },
          {
            "schema": {
              "type": "object",
              "required": ["data"],
              "properties": {
                "data": {
                  "type": "object",
                  "required": ["example_two"],
                  "properties": {
                    "example_two": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        ]
      }
    }
  },
  "$schema": "http://json-schema.org/draft-04/schema#"
}
```
- While bundling will result in
```json
{
  "components": {
    "main": {
      "schema": {
        "discriminator": {
          "propertyName": "key"
        },
        "properties": {
          "key": {
            "type": "string",
            "enum": ["value1", "value2"]
          }
        },
        "anyOf": [
          { "$ref": "#components/value1" },
          { "$ref": "#components/value2" }
        ]
      }
    },
    "value1": {
      "schema": {
        "type": "object",
        "required": ["data"],
        "properties": {
          "data": {
            "type": "object",
            "required": ["example_one"],
            "properties": {
              "example_one": {
                "type": "integer",
                "exclusiveMinimum": 0
              }
            }
          }
        }
      }
    },
    "value2": {
      "schema": {
        "type": "object",
        "required": ["data"],
        "properties": {
          "data": {
            "type": "object",
            "required": ["example_two"],
            "properties": {
              "example_two": {
                "type": "string"
              }
            }
          }
        }
      }
    }
  },
  "$schema": "http://json-schema.org/draft-04/schema#"
}
```

However, changing this requires us to change all of our schemas to proper OpenAPI format, as well as implement proper bundling behavior (which is not well-defined right now). I think we may need to avoid using `discriminator` and rely on a custom schema for now. We'll think of a path to update our existing shemas to be more inline with the opensource tools.